crates/fuel-core/src/coins_query.rs

```diff
@@ -221,6 +279,222 @@ pub async fn random_improve(
     Ok(coins_per_asset)
 }
 
+pub async fn select_coins_to_spend(
+    CoinsToSpendIndexIter {
+        big_coins_iter,
+        dust_coins_iter,
+    }: CoinsToSpendIndexIter<'_>,
+    total: u64,
+    max: u16,
+    asset_id: &AssetId,
+    excluded_ids: &ExcludedCoinIds<'_>,
+    batch_size: usize,
+) -> Result<Vec<CoinsToSpendIndexEntry>, CoinsQueryError> {
+    const TOTAL_AMOUNT_ADJUSTMENT_FACTOR: u64 = 2;
```

Comment:
It would be nice to put a comment why we decided to do it like this=)

Reply:
Added in d9d024bd16777cb2a9468fb616b273b15b3e1fbe